cmd/babylond/cmd/create_bls_key.go

```diff
@@ -27,7 +29,38 @@ $ babylond create-bls-key --home ./
 
 		RunE: func(cmd *cobra.Command, args []string) error {
 			homeDir, _ := cmd.Flags().GetString(flags.FlagHome)
-			appsigner.GenBls(appsigner.DefaultBlsKeyFile(homeDir), appsigner.DefaultBlsPasswordFile(homeDir), blsPassword(cmd))
+			noBlsPassword, _ := cmd.Flags().GetBool(flagNoBlsPassword)
+
+			blsKeyFile := appsigner.DefaultBlsKeyFile(homeDir)
+			blsPasswordFile := appsigner.DefaultBlsPasswordFile(homeDir)
+
+			if err := appsigner.EnsureDirs(blsKeyFile, blsPasswordFile); err != nil {
+				return fmt.Errorf("failed to ensure dirs exist: %w", err)
+			}
+
+			if noBlsPassword {
+				bls := appsigner.NewBls(bls12381.GenPrivKey(), blsKeyFile, blsPasswordFile)
+				bls.Key.Save("")
+				fmt.Printf("BLS key generated successfully without password protection.\n")
+				return nil
+			}
+
+			password, _ := cmd.Flags().GetString(flagBlsPassword)
+			if password == "" {
+				password = appsigner.NewBlsPassword()
+			}
+
+			bls := appsigner.NewBls(bls12381.GenPrivKey(), blsKeyFile, blsPasswordFile)
+			bls.Key.Save(password)
```

Comment:
Seems we still store the password in the file? I thought our intention is not to store the password anywhere and let user decide how to manage the password?